aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java

```diff
@@ -35,18 +36,19 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(final Amaz
                                                                           final CallbackContext callbackContext,
                                                                           final ProxyClient<RdsClient> proxyClient,
                                                                           final Logger logger) {
-        final Tagging.TagSet systemTags = Tagging.TagSet.builder()
-                .systemTags(Tagging.translateTagsToSdk(request.getSystemTags()))
-                .build();
 
         final Tagging.TagSet extraTags = Tagging.TagSet.builder()
                 .stackTags(Tagging.translateTagsToSdk(request.getDesiredResourceTags()))
                 .resourceTags(Translator.translateTagsToSdk(request.getDesiredResourceState().getTags()))
                 .build();
 
+        final Tagging.TagSet allTags = extraTags.toBuilder()
+                .systemTags(Tagging.translateTagsToSdk(request.getSystemTags()))
+                .build();
+
         return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
                 .then(progress -> setDbClusterParameterGroupNameIfMissing(request, progress))
-                .then(progress -> createDbClusterParameterGroup(proxy, proxyClient, progress, systemTags))
+                .then(progress -> createDbClusterPGWithAllTagsOrSystemTags(proxy, proxyClient, progress, allTags))
```

Comment:
nit: too much implementation details in the method name. Consider: `safeCreateDBClusterParameterGroup`

Reply:
Done! I updated method logic and naming. Please check if you have better naming.